test/e2e/util_test.go

```diff
@@ -598,31 +599,117 @@ func updateIngressConfigSpecWithRetryOnConflict(t *testing.T, name types.Namespa
-		if err := kclient.Get(context.TODO(), name, &infraConfig); err != nil {
+	// First we apply the configuration
+	err := wait.PollUntilContextTimeout(context.TODO(), 1*time.Second, timeout, false, func(ctx context.Context) (done bool, err error) {
+		if err := kclient.Get(ctx, name, &infraConfig); err != nil {
 			t.Logf("error getting infrastructure config %v: %v, retrying...", name, err)
 			return false, nil
 		}
 		mutateSpecFn(&infraConfig.Spec)
 		if err := kclient.Update(context.TODO(), &infraConfig); err != nil {
-			if errors.IsConflict(err) {
-				t.Logf("conflict when updating infrastructure config %v: %v, retrying...", name, err)
+			t.Logf("error updating infrastructure config %v: %v, retrying...", name, err)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+	// Then we get the applied configuration, and check if the serviceendpoints match
+	return wait.PollUntilContextTimeout(context.TODO(), 1*time.Second, timeout, false, func(ctx context.Context) (done bool, err error) {
+		if err := kclient.Get(ctx, name, &infraConfig); err != nil {
+			t.Logf("error getting infrastructure config %v: %v, retrying...", name, err)
+			return false, nil
+		}
+
+		// Early return if we don't have any specific service endpoint
+		if infraConfig.Spec.PlatformSpec.AWS == nil || len(infraConfig.Spec.PlatformSpec.AWS.ServiceEndpoints) == 0 {
+			return true, nil
+		}
+		// In case we have a desired ServiceEndpoints, we need to check once it has been reflected correctly
+		if infraConfig.Status.PlatformStatus == nil ||
+			infraConfig.Status.PlatformStatus.AWS == nil ||
+			len(infraConfig.Status.PlatformStatus.AWS.ServiceEndpoints) != len(infraConfig.Spec.PlatformSpec.AWS.ServiceEndpoints) {
+			t.Logf("infrastructure config status does not contain endpoints yet %v: %v, retrying...", name, infraConfig.Status)
+			return false, nil
+		}
+
+		for _, endpoint := range infraConfig.Spec.PlatformSpec.AWS.ServiceEndpoints {
+			if !slices.ContainsFunc(infraConfig.Status.PlatformStatus.AWS.ServiceEndpoints, func(e configv1.AWSServiceEndpoint) bool {
+				return e.Name == endpoint.Name && e.URL == endpoint.URL
+			}) {
+				t.Logf("infrastructure config status does not contain the desired endpoints, retrying %v: %v, retrying...", name, endpoint)
 				return false, nil
 			}
-			return false, err
 		}
 		return true, nil
 	})
 }
 
+// waitForCCMReadiness waits for Cloud Controller Manager to be ready.
+//
+//	for infrastructure status to update with custom endpoints,
+//
+// for CCM deploy to get a new configuration hash and for pods to be scheduled
+// and available
+// This configuration change and replicas available may take up to 5 minutes, depending on
+// the environment load
+// If oldConfigurationHash is not empty it will additionally verify that the configuration hash
+// is different between the deployments
+func waitForCCMReadiness(t *testing.T, timeout time.Duration, oldConfigurationHash string) appsv1.Deployment {
```

Comment:
We don't just "wait" in here but we assert that the CCM is ready (`t.Fatal` if not). This should be reflected in the function name.

Reply:
Renamed for `waitForCCMAvailableAndUpdated`